routes/duties.js

```diff
@@ -0,0 +1,141 @@
+import {
+  createNewDuty, lookForAllDuties, lookForDutyById, deleteDutyById, updateDuty,
+} from '../database.js';
+
+const newDutySchema = {
+  schema: {
+    body: {
+      type: 'object',
+      additionalProperties: false,
+      required: ['name', 'location', 'time', 'constraints', 'soldiersRequired', 'value'],
+      properties: {
+        name: { type: 'string' },
+        location: { type: 'string' },
+        time: {
+          type: 'object',
+          properties: {
+            start: { type: 'string', format: 'date' },
+            end: { type: 'string', format: 'date' },
+          },
+        },
+        constraints: { type: 'array' },
+        soldiersRequired: { type: 'number' },
+        value: { type: 'number' },
+      },
+
+    },
+  },
+
+};
+
+const getDutySchema = {
+  schema: {
+    query: {
+      type: 'object',
+      properties: {
+        name: { type: 'string' },
+        location: { type: 'string' },
+        time: {
+          type: 'object',
+          properties: {
+            start: { type: 'string', format: 'date' },
+            end: { type: 'string', format: 'date' },
+          },
+        },
+        constraints: { type: 'array' },
+        soldiersRequired: { type: 'number' },
+        value: { type: 'number' },
+      },
+    },
+  },
+
+};
+
+const dutyEditSchema = {
+  type: 'object',
+  properties: {
+    name: { type: 'string' },
+    location: { type: 'string' },
+    time: {
+      type: 'object',
+      properties: {
+        start: { type: 'string', format: 'date' },
+        end: { type: 'string', format: 'date' },
+      },
+    },
+    constraints: { type: 'array' },
+    soldiersRequired: { type: 'number' },
+    value: { type: 'number' },
+  },
+
+};
+
+export default function dutyRouter(app, opts, done) {
+  app.post('/', newDutySchema, async (req, res) => {
+    const dutyInserted = req.body;
+    dutyInserted._id = Math.floor(Math.random() * 1000000);
+    await createNewDuty(dutyInserted);
+    res.status(201).send({ message: 'duty created' });
+  });
+
+  app.get('/', getDutySchema, async (req, res) => {
+    const dutiesQuery = req.query;
+    const searchResult = await lookForAllDuties(dutiesQuery);
+    res.send(searchResult);
+  });
+
+  app.get('/:id', async (req, res) => {
+    const dutyId = req.params.id;
+    const searchResponse = await lookForDutyById(
+      { _id: Number(dutyId) },
+    );
+    if (searchResponse) {
+      res.status(200).send({ message: 'duty found' });
+    } else {
+      res.status(400).send({ message: 'duty not found' });
+    }
+  });
+
+  app.delete('/:id', async (req, res) => {
+    const idForDelete = req.params.id;
+    const searchResponse = await lookForDutyById(
+      { _id: Number(idForDelete) },
+    );
+    if (searchResponse) {
```

Comment:
if you like you can use the negative option and then return early. this way you don't need to nest code:
 ```suggestion
    if (!searchResponse) return res.status(400).send({ message: 'duty doesnt exist' });
    if (!searchResponse.soldiers.length) return res.status(400).send({ message: 'scheduled duties cannot be deleted' });
    // continue as usual
```

Reply:
fixed